src/terminal.rs

```diff
@@ -61,28 +63,40 @@ fn set_flags(fd: RawFd, flags: c_int) -> Result<()> {
     Ok(())
 }
 
-/// Trait for file descriptors that are TTYs, according to `isatty(3)`.
+/// Trait for file descriptors that are TTYs, according to
+/// [`isatty`](http://man7.org/linux/man-pages/man3/isatty.3.html).
 ///
-/// This is marked unsafe because the implementation must promise that the returned RawFd is a valid
-/// fd and that the lifetime of the returned fd is at least that of the trait object.
+/// This is marked unsafe because the implementation must promise that the returned
```

Comment:
Replace "must promise" with "must ensure"

Reply:
Thanks for the review. Updated. :)